sw/device/lib/drivers/i2c/i2c.c

```diff
@@ -1,102 +1,120 @@
-// Copyright lowRISC contributors.
-// Licensed under the Apache License, Version 2.0, see LICENSE for details.
-// SPDX-License-Identifier: Apache-2.0
+/*
+                              *******************
+******************************* C SOURCE FILE *******************************
+**                            *******************                          **
+**                                                                         **
+** project  : x-heep                                                       **
+** filename : i2c.c                                                        **
+** date     : 16/05/2023                                                   **
+**                                                                         **
+*****************************************************************************
+**                                                                         **
+**                                                                         **
+*****************************************************************************
+	 
+*/
+
+/***************************************************************************/
+/***************************************************************************/
+
+/**
+* @file   i2c.c
+* @date   17/04/2023
+* @brief  This is the main file for the HAL of the I2C peripheral
+*
+* In this files there are definitions of low level HAL functions to interact
+* with the registers of the I2C peripheral.
+* The functionalities implemented allow to configure the peripheral with
+* parameters for the FMT and RX fifos, enable or disable interrupts and write
+* and read bytes.
+*/
+
+
+/****************************************************************************/
+/**                                                                        **/
+/*                             MODULES USED                                 */
+/**                                                                        **/
+/****************************************************************************/
 
 #include "i2c.h"
 
+#include "i2c_regs.h"
+#include "i2c_structs.h"
+
 #include "bitfield.h"
-#include "i2c_regs.h"  // Generated
+
+/****************************************************************************/
+/**                                                                        **/
+/*                        DEFINITIONS AND MACROS                            */
+/**                                                                        **/
+/****************************************************************************/
+
+static const uint32_t kNanosPerKBaud = 1000000;  // One million.
```

Comment:
what is this? this is neither a definition nor a macro, place it into the global variables if only employed in the scope of this file, plus, add some comments explaining what it is and why it is init to that value